### ISSUE TITLE
input from stdin or a file

### ISSUE THREAD

**Issue by windrg** [rizsotto/Bear#456]
**Describe the bug**
In Usage, I cannot find a way to input from STDIN or a file - ie build.log

**To Reproduce**
Steps to reproduce the behavior: <not exist>

**Expected behavior**
Other tools provides a way to do this. But Bear doesn’t seem to have.

**Environment:**
 - OS name: Ubuntu
 - OS version: 20.04 
 - OS architecture: x86_64
 - Bear version: <not checked>
 - Bear install method: apt
**Additional context**
 - Can you give us a reference to the project that you are running against this tool? [e.g. No if it's a close source project, but in case of an open source project a link to the sources could be nice.]
    >> compiledb can get inputs from stdin or a file
 - What build tools this project is using? [e.g. GNU make wtih `configure` script]
    >> make
 - What architecture you are trying to compile for? [e.g. cross compiling]
    >> ARM - cross compiling
 - Could you attach build logs? If you can re-run the command which failed (e.g. `bear -- make`) with extra verbose log switches (e.g. `bear --verbose -- make`) and send the output of it, that would help a lot.
    >> No, I can’t. But Bear works when I issue like this
          $bear make ….
          Then it generates a compile_commands.json very well as I expected.

**Comment by rizsotto** [comment]
Bear does not process build logs, therefore it won't be a thing.

Bear **runs** the build and intercepts the process executions. The program `intercept` is which generates the event log file about the intercepted process execution. And `citnames` is another tool, which reads this event log file and creates the compilation database.

**Comment by rizsotto** [comment]
@windrg - thanks for the request.

On the `4.2.0-rc` branch, `bear parse-sh` reads shell text from stdin or a
file and writes the event stream to stdout or a file, so it's a plain filter:

```sh
cat build.log | bear parse-sh | bear semantic --input -
# or with files:
bear parse-sh -i build.log -o events.json
```

- your log (or `make -n` output) is the input text;
- `bear parse-sh` turns each command line into an execution event;
- `bear semantic --input -` writes `compile_commands.json`.

Two things that may help your case: the source files named in the log don't
need to exist on the machine running `parse-sh`, and `--directory` lets you
pin the build directory for a log captured elsewhere. A full build log is
best-effort though - only lines that are shell commands are parsed, and
interleaved compiler output is skipped (see #287).

It's not released yet - if you can build the `4.2.0-rc` branch and run one of
your logs through it, I'd love to hear how much it recovers.
